### ISSUE TITLE
How to use?

### ISSUE THREAD

**Issue by Mansoun** [daveoggy/Dapper.AutoRefresh#1]
Hi. Is there any example to how use this library?
thanks.

**Comment by oggy-bh** [comment]
It's been a while since I thought about this library. I don't think the required components (SqlDependency) exist in .NET core. So this library only works with .NET Framework. Given that limitation is this still of interest?

**Comment by Mansoun** [comment]
Hi again. 
Unfortunately our project developing in .net core . My problem solved with classic Ado.net .

**Comment by dhienoz** [comment]
So is there a sample code how to use this library as previously being asked? .NET framework is fine.

**Comment by daveoggy** [comment]
I'll update the project with an example. Give me a couple of days.

**Comment by dhienoz** [comment]
That would be awesome. Thank you!

**Comment by daveoggy** [comment]
Done. Sorry for the delay. Let me know how you get on.

**Comment by dhienoz** [comment]
Thank you for providing the sample code. I find it very helpful.
I'd like to hook up an event to the SqlDependency's OnChange event. This event will get passed from the SqlAutoRefresh's constructor, something like this:

var sqlAutoRefresh = new SqlAutoRefresh<MyType>("SELECT Id, Name FROM dbo.MyType", onChangeEvent, connectionSB.ToString());

public SqlSelectRefresh(string sqlQuery,
                        OnChangeEventHandler onChangeEvent,
			string connectionString,
			...)
		{
                        ...
			_sqlDependencyAsync = new SqlDependencyAsync(_cancellationToken, onChangeEvent);
                        ...
		}

public SqlDependencyAsync(OnChangeEventHandler onChangeEvent)
{
     SqlDependency.OnChange += onChangeEvent;
}

That would work don't you think?

**Comment by daveoggy** [comment]
It's seems you might be in this situation: https://xyproblem.info

Do you want to tell me what it is you're trying to do?

**Comment by dhienoz** [comment]
As briefly mentioned before, I want to associate an event to SqlDependency's OnChange event using Dapper.
Based on this particular sample code, when a new record is inserted/updated, I'd like to have that OnChange event fired (which would call up the event that I pass along to the SqlDependencyAsync). 
Does that make sense?

**Comment by daveoggy** [comment]
I don't think you need to worry about the OnChange event. This library has abstracted that away for you. From the example code - If you want to know when data related to the query you use has changed, that's when the `var data = await sqlAutoRefresh.GetLatest();` is allowed to continue. The variable `data` will then already have the latest data from the table automatically.

If you're trying to do something other than get the latest dataset when the table changes, this library might not be for you.
